public/components/integrations/components/setup_integration.tsx

```diff
@@ -115,201 +166,120 @@ export function IntegrationDataModal({ close }: { close: () => void }): React.JS
   );
 }
 
-export function SetupIntegrationNewTable({
+const findTemplate = async (integrationTemplateId: string) => {
+  const http = coreRefs.http!;
+  const result = await http.get(`${INTEGRATIONS_BASE}/repository/${integrationTemplateId}`);
+  return result;
+};
+
+export function SetupIntegrationForm({
   config,
   updateConfig,
-}: {
-  config: IntegrationConfig;
-  updateConfig: (updates: Partial<IntegrationConfig>) => void;
-}) {
+  integration,
+}: IntegrationConfigProps) {
+  const connectionType =
+    INTEGRATION_CONNECTION_DATA_SOURCE_TYPES.get(config.connectionType) ?? 'index';
+  const indefiniteArticle = 'aeiou'.includes(connectionType.charAt(0)) ? 'an' : 'a';
+  const capitalizedConnectionType =
+    connectionType.charAt(0).toUpperCase() + connectionType.slice(1);
+
+  const [dataSourceSuggestions, setDataSourceSuggestions] = useState(
+    [] as Array<{ label: string }>
+  );
+  const [isSuggestionsLoading, setIsSuggestionsLoading] = useState(true);
+  useEffect(() => {
+    const updateDataSources = async () => {
+      const data = await suggestDataSources(config.connectionType);
+      setDataSourceSuggestions(data);
+      setIsSuggestionsLoading(false);
+    };
+
+    setIsSuggestionsLoading(true);
+    updateDataSources();
+  }, [config.connectionType]);
+
   return (
-    <div>
-      <Eui.EuiFormRow label="Title">
-        <Eui.EuiFieldText
-          value={config.dataSourceName}
-          onChange={(evt) => updateConfig({ dataSourceName: evt.target.value })}
-        />
-      </Eui.EuiFormRow>
-      <Eui.EuiFormRow label="Description (optional)">
+    <Eui.EuiForm>
```

Comment:
I think convention for this plugin is to just import the component directly and use it and not use dot notation here

Reply:
Yeah but I didn't like having 40ish lines of imports